packages/paste-core/components/toast/src/types.ts

```diff
@@ -8,19 +8,47 @@ export interface ToastProps extends React.HTMLAttributes<HTMLDivElement> {
   children: NonNullable<React.ReactNode>;
   onDismiss?: () => void;
   variant: ToastVariants;
+  setFocus?: boolean;
 }
-
-export interface ToastContainerProps extends React.HTMLAttributes<HTMLDivElement> {
-  children: NonNullable<React.ReactNode>;
-}
-
 export const ToastPropTypes = {
   children: PropTypes.node.isRequired,
   onDismiss: PropTypes.func,
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   variant: PropTypes.oneOf(['error', 'neutral', 'success', 'warning']) as any,
 };
 
+export interface ToastContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  children: NonNullable<React.ReactNode>;
+}
 export const ToastContainerPropTypes = {
   children: PropTypes.node.isRequired,
 };
+
+export interface ToasterToast {
+  id?: string;
+  message: React.ReactElement | string;
+  dismissAfter?: number;
+  timeOutId?: number;
+  setFocus?: boolean;
+  variant: ToastVariants;
+}
+
+export interface UseToasterReturnedProps {
+  toasts: ToasterToast[];
+  push: (toast: ToasterToast) => void;
+  pop: (id: ToasterToast['id']) => void;
+}
+
+export type ToasterProps = Pick<UseToasterReturnedProps, 'toasts' | 'pop'>;
+export const ToasterPropTypes = {
+  toasts: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      message: PropTypes.node.isRequired,
+      dismissAfter: PropTypes.number,
+      timeOutId: PropTypes.number,
```

Comment:
Looks like these two types aren't used here, only on "ToasterToast"

Reply:
Well toasts is an array of `ToasterToast`. They are valid if you're _not_ using the `useToaster` hook and for what ever reason you are wanting to roll your own state. In which case you're dealing with setting this stuff